### PR TITLE
Fixes an issue of GetBayesianHetCoverage with non- ACTG (e.g. N, X, etc) reference bases

### PR DESCRIPTION
**Pull request by mbabadi**
... by filtering out non-ACTG bases when (1) walking the reference and (2) making a pileup.

### CODE REVIEW

**Comment by mbabadi** [comment]
@LeeTL1220 I disabled indexing in SamLocusIterator by default. In GetHetCoverage, it is pretty much always disabled (it is enabled only if #SNPS < 25000). There must a bug in SamLocusIterator since indexing leads to a neverending iterator sometimes. I found another example. Could you check if this PR fixes your problematic case as well?

**Comment by LeeTL1220** [comment]
@mbabadi ok... I'll get to it asap....

**Comment by LeeTL1220** [comment]
@mbabadi Hey, as a reminder, I have this bug for both GetBayesian and regular GetHetCoverage...

**Comment by LeeTL1220** [comment]
@mbabadi In other words, I am not going to test this, since you made GetBayesian more like GetHet

File: src/main/java/org/broadinstitute/hellbender/tools/exome/pulldown/BayesianHetPulldownCalculator.java [hunk 1]
```diff
@@ -184,6 +184,20 @@ private double getSingletonLogLikelihood(final Map<Nucleotide, List<BaseQuality>
...
+        for (final Nucleotide properBase : BASES) {
+            if (base.equals(properBase)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
      * Returns map of base-pair to error probabilities at a given locus. All reads are considered (not just ACTG)
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/pulldown/BayesianHetPulldownCalculator.java]: Is this comment still applicable?

**Comment by samuelklee** [comment]
Not sure why this was assigned to me---should I review as well?

**Comment by LeeTL1220** [comment]
@mbabadi Not much.... but give it a look.  Feel free to merge after that.

@samuelklee whoops!

File: src/main/java/org/broadinstitute/hellbender/tools/exome/pulldown/BayesianHetPulldownCalculator.java (hunk 1 above)
Comment (mbabadi) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/pulldown/BayesianHetPulldownCalculator.java]: It is still applicable. The improper base calls, however, will be ignored in the inference.

**Pull request merged**